perfkitbenchmarker/providers/ibmcloud/ibmcloud_disk.py

```diff
@@ -0,0 +1,199 @@
+from perfkitbenchmarker import vm_util
+
+from perfkitbenchmarker.providers.ibmcloud import ibm_api as ibm
+
+FLAGS = flags.FLAGS
+
+ATTACHED = 'attached'
+AVAILABLE = 'available'
+DEFAULT_DISK_SIZE = 1000
+
+
+class IbmCloudDisk(disk.BaseDisk):
+  """Object representing a IBM Cloud Disk"""
+
+  def __init__(self, disk_spec, name, zone):
+    super(IbmCloudDisk, self).__init__(disk_spec)
+    self.name = name
+    self.zone = zone
+    self.vol_id = None
+    self.attached_vm = None
+    self.attached_vdisk_URI = None
+    self.device_path = None
+    self.data_disk_size = FLAGS.data_disk_size if FLAGS.data_disk_size is not None else DEFAULT_DISK_SIZE
+
+  def _Create(self):
+    """Creates an external block volume."""
+    volcmd = ibm.IbmAPICommand(self)
+    volcmd.flags['name'] = self.name
+    volcmd.flags['zone'] = self.zone
+    volcmd.flags['iops'] = FLAGS.ibmcloud_volume_iops
+    volcmd.flags['profile'] = FLAGS.ibmcloud_volume_profile
+    volcmd.flags['capacity'] = self.data_disk_size
+    if FLAGS.ibmcloud_datavol_encryption_key:
+      volcmd.flags['encryption_key'] = FLAGS.ibmcloud_datavol_encryption_key
+    if FLAGS.ibmcloud_rgid:
+      volcmd.flags['resource_group'] = FLAGS.ibmcloud_rgid
+
+    logging.info('Volume create, volcmd.flags %s', volcmd.flags)
+    resp = json.loads(volcmd.CreateVolume())
+    self.vol_id = resp['id']
+    logging.info('Volume id: %s', self.vol_id)
+
+  def _Delete(self):
+    """Deletes an external block volume."""
+    if self.vol_id is None:
+      logging.info('Volume %s was not created. Skipping deletion.' % self.name)
+      return
+    volcmd = ibm.IbmAPICommand(self)
+    volcmd.flags['volume'] = self.vol_id
+    logging.info('Volume delete, volcmd.flags %s', volcmd.flags)
+    volcmd.DeleteVolume()
+    logging.info('Volume deleted: %s', self.vol_id)
+    self.vol_id = None
+
+  def _Exists(self):
+    if self.vol_id is None:
+      return False
+    return True
+
+  def _FindVdiskIdByName(self, vdisks, name):
+    """Finds vdisk id by name"""
+    for vdisk in vdisks:
+      if vdisk['name'] == name:
+        return vdisk['id']
+    return None
+
+  def Attach(self, vm):
+    """Attaches the disk to a VM.
+
+    Args:
+      vm: instance of the vm to which the disk will be attached.
+    """
+    if self.vol_id is None:
+      raise errors.Error('Cannot attach remote volume %s' % self.name)
+    if vm.vmid is None:
+      msg = 'Cannot attach remote volume %s to non-existing %s VM' % (self.name,
+                                                                      vm.name)
+      raise errors.Error(msg)
+
+    volcmd = ibm.IbmAPICommand(self)
+    volcmd.flags['name'] = self.name
+    volcmd.flags['instanceid'] = vm.vmid
+    volcmd.flags['volume'] = self.vol_id
+    volcmd.flags['delete'] = True
+
+    # wait till volume is PROVISIONED before attach
+    resp = json.loads(volcmd.ShowVolume())
+    status = resp['status']
+    endtime = time.time() + 300
```

Comment:
Can you make this a constant?  In this case at the top of the file:
  _MAX_DISK_ATTACH_SECONDS = 300

endtime = time.time() + _MAX_DISK_ATTACH_SECONDS

Reply:
yes, added constants.